### PR TITLE
Feature/configure max voices

### LINKED ISSUE #1803: Maximum polyphony per synth is not configurable
### Please describe the problem:

The maximum polyphony for a track is currently either 16 (polyphonic) or 1 (monophonic/legato)

The check is here https://github.com/m-m-adams/DelugeFirmware/blob/be97feb547c92a6e9620bf34b05ee3dce9cca5b2/src/deluge/processing/engines/audio_engine.cpp#L1319 

### What is the expected behavior?

It would be nice if this was a user configurable option in the polyphony menu, at least something like 1/4/8/16

### Is there a relevant Pull request?

#1751

### What hardware did you reproduce it with?

OLED, 7SEG

### What firmware did you reproduce it with?

Nightly, other

### What is the firmware name:

_No response_

### If possible provide the steps to reproduce the issue and upload additional media:

_No response_

### PR DESCRIPTION
**Pull request by m-m-adams**
Close #1803 

Adds a submenu under the polyphonic option in the polyphony menu to set a max number of voices. It defaults to 8 for new synths and 16 for old synths where a max number is not configured.

Also makes more enums single byte, poor ordering and large enums was making sounds and voices ~20 bytes larger than needed

Fixes the IntegerWithOff class to also show off on 7seg - affects the MPE channel count menu, MPE velocity menu, count in bars, and Saturation menus.

### CODE REVIEW

File: src/deluge/gui/l10n/english.json [hunk 1]
```diff
@@ -1,809 +1,731 @@
...
+    "STRING_FOR_TIME": "Time",
+    "STRING_FOR_MASTER": "Master",
+    "STRING_FOR_FILTERS": "Filters",
+    "STRING_FOR_CANT_TRANSPOSE": "Can't Transpose: No Scale Mode Clips",
+    "STRING_FOR_TRANSPOSE_INKEY": "In Key",
+    "STRING_FOR_TRANSPOSE_CHROMATIC": "Chromatic",
+    "STRING_FOR_TRANSPOSE_CHORD": "Chord",
+    "STRING_FOR_CANT_ENTER_SCALE": "Can't enter scale mode, MIDI transpose is chromatic",
+    "STRING_FOR_DEFAULT_HIGH_CPU_USAGE_INDICATOR": "High CPU Indicator"
+  }
```
Comment (sapphire-arches) [inline src/deluge/gui/l10n/english.json:729]: Something weird happened here, maybe line endings?

File: src/deluge/processing/sound/sound.cpp [hunk 2]
```diff
@@ -1253,6 +1253,11 @@ Error Sound::readTagFromFile(StorageManager& bdsm, char const* tagName, ParamMan
 		bdsm.exitTag("polyphonic");
 	}
 
+	else if (!strcmp(tagName, "maxVoices")) {
+		maxVoiceCount = bdsm.readTagOrAttributeValueInt();
+		bdsm.exitTag("maxVoices");
+	}
+
```
Comment (sapphire-arches) [inline src/deluge/processing/sound/sound.cpp:1260]: Does this cause official to crash/not load subsequent settings?

**Comment by sapphire-arches** [comment]
It would probably be good to have the voice count directly accessible under the `VOICE` menu as well -- honestly I might prefer that over the select-swap menu thing? That interaction pattern is already used for unison stereo spread but it feels kind of weird.

**Comment by m-m-adams** [comment]
> It would probably be good to have the voice count directly accessible under the `VOICE` menu as well -- honestly I might prefer that over the select-swap menu thing? That interaction pattern is already used for unison stereo spread but it feels kind of weird.

Yeah I'll add it to the voice menu too. I like having it in the polyphony menu as well since it's semantically choosing "how polyphonic" and that gives a way to access it via shortcuts

File: src/deluge/gui/l10n/english.json (hunk 1 above)
Comment (m-m-adams) [reply src/deluge/gui/l10n/english.json:729]: Oops yeah - I'll check

File: src/deluge/processing/sound/sound.cpp (hunk 2 above)
Comment (m-m-adams) [reply src/deluge/processing/sound/sound.cpp:1260]: yep, moved to the end

File: src/deluge/processing/sound/sound.cpp [hunk 3]
```diff
@@ -4022,6 +4021,9 @@ void Sound::writeToFile(StorageManager& bdsm, bool savingSong, ParamManager* par
 	bdsm.writeClosingTag("modKnobs");
 
 	ModControllableAudio::writeTagsToFile(bdsm);
+	
+	// community attributes, new additions will stop official firmware from continuing to parse the sound
+	bdsm.writeAttribute("maxVoices", maxVoiceCount);
```
Comment (soymonitus) [inline src/deluge/processing/sound/sound.cpp:4026]: Yes! glad you remembered it :)

Comment (m-m-adams) [reply src/deluge/processing/sound/sound.cpp:4026]: Well Reed did at least!

**Comment by seangoodvibes** [comment]
This is a wonderful addition! Seems to work great!

**Pull request merged**